system-int-test/test/cmr/system_int_test/search/tool/tool_search_test.clj

```diff
@@ -0,0 +1,523 @@
+    (index/wait-until-indexed)
+
+    (are3 [expected-tools keyword-query]
+      (d/assert-refs-match
+       expected-tools (tool/search-refs {:keyword keyword-query}))
+
+      "Tool Category"
+      [tl1]
+      "tool category"
+
+      "Tool Specific Term"
+      [tl1]
+      "specific tool term"
+
+      "Tool Term"
+      [tl1]
+      "tool term"
+
+      "Tool Topic"
+      [tl1]
+      "tool topic")))
+
+(deftest search-organization-keywords-test
+  (let [tl1 (tool/ingest-tool-with-attrs {:native-id "tl-1"
+                                          :Name "Tool 1"
+                                          :Organizations [(data-umm-t/organization)]})
+        tl2 (tool/ingest-tool-with-attrs {:native-id "tl-2"
+                                          :Name "Tool 2"})]
+    (index/wait-until-indexed)
+
+    (are3 [expected-tools keyword-query]
+      (d/assert-refs-match
+       expected-tools (tool/search-refs {:keyword keyword-query}))
+
+      "Short Name"
+      [tl1]
+      "tlorg1"
+
+      "Long Name"
+      [tl1]
+      "Tool Org 1"
+
+      "Roles"
+      [tl1 tl2]
+      "SERVICE PROVIDER")))
+
+(deftest tool-search-sort
+  (let [tool1 (tool/ingest-tool-with-attrs {:native-id "tl1"
+                                            :Name "tool"
+                                            :LongName "LongName4"
+                                            :provider-id "PROV2"})
+        tool2 (tool/ingest-tool-with-attrs {:native-id "tl2"
+                                            :Name "Tool 2"
+                                            :LongName "LongName2"
+                                            :provider-id "PROV1"})
+        tool3 (tool/ingest-tool-with-attrs {:native-id "tl3"
+                                            :Name "a tool"
+                                            :LongName "LongName1"
+                                            :provider-id "PROV1"})
+        tool4 (tool/ingest-tool-with-attrs {:native-id "tl4"
+                                            :Name "tool"
+                                            :LongName "LongName3"
+                                            :provider-id "PROV1"})]
+    (index/wait-until-indexed)
+
+    (are3 [sort-key expected-tools]
+      (is (d/refs-match-order?
+           expected-tools
+           (tool/search-refs {:sort-key sort-key})))
+
+      "Default sort"
+      nil
+      [tool3 tool4 tool1 tool2]
+
+      "Sort by name"
+      "name"
+      [tool3 tool4 tool1 tool2]
+
+      "Sort by name descending order"
+      "-name"
+      [tool2 tool4 tool1 tool3]
+
+      "Sort by provider id"
+      "provider"
+      [tool2 tool3 tool4 tool1]
+
+      "Sort by provider id descending order"
+      "-provider"
+      [tool1 tool2 tool3 tool4]
+
+      "Sort by revision-date"
+      "revision_date"
+      [tool1 tool2 tool3 tool4]
+
+      "Sort by revision-date descending order"
+      "-revision_date"
+      [tool4 tool3 tool2 tool1]
```

Comment:
These two revision-date sorting tests are really strange:
1. It failed consistently, locally if I ran the whole test suite.(even after I added delays between each ingest).
2. It succeeded consistently, locally if I ran only tool_search_test.clj
3. It succeeded in bamboo.